InterGalacticCurrencyConvertor/src/main/java/materials/Materials.java

```diff
@@ -0,0 +1,57 @@
+package main.java.materials;
+
+import java.util.HashMap;
+import java.util.Objects;
+
+/**
+ * Keeps track of different types of materials that can be sold InterGalactically.
+ */
+public class Materials {
+	private HashMap<String, Material> mMap  = new HashMap<>();
+	private static Materials mInstance = new Materials();
+	public static Materials getInstance() {
+		return mInstance;
+	}
+	public void addMaterial(Material inMaterial) {
+	  Objects.requireNonNull(inMaterial, "Invalid Material Name");
+		mMap.put(inMaterial.getMaterialName(), inMaterial);
+	}
+	public boolean isMaterialAvailable(String inMaterialName) {
+		return mMap.containsKey(inMaterialName);
+	}
+	public double getMaterialUnitPrice(String inMaterialName) {
+	  Objects.requireNonNull(inMaterialName, "Invalid Material Name");
+		if (!isMaterialAvailable(inMaterialName)) {
+			throw new IllegalArgumentException("Invalid Material Name provide");
+		}
+		return mMap.get(inMaterialName).getMaterialUnitPrice();
+	}
+	public void createMaterial(String inMaterialName,
```

Comment:
Remove this method - it is confusing. Does it just create? Or does it create and add? Let the caller create and add in two steps.

Reply:
removed